### ISSUE TITLE
Cannot submit requested media information to complete analysis

### ISSUE THREAD

**Issue by woutervanrooy** [ebu/pi-list#85]
Hello,

I am using the LIST tool for analysis of some ST2110-30 packet captures. For some of these captures, there are multiple combinations of encoding, channel count and packet time that match the packet size. LIST will then ask for additional media information so that it can complete the analysis. So far, so good.

However, whenever I try to submit one of the suggested compliant profiles the system runs into an internal server error:

> 2020-06-09T10:49:43.945Z [pcap-full-analysis] error: exception: Error: Command failed: "/app/bin//st2110_extractor" 7908c5d0-aa38-11ea-88d4-e341cbc94286 "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.pcap" "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286" -influx_url http://influxdb:8086 -mongo_url mongodb://mongo:27017 -s 1f487b99-8708-4d34-bcce-c0921612dcc7 -p /home/5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/profile.json 
2020-06-09T10:49:43.950Z [express-middleware] error: Error: Command failed: "/app/bin//st2110_extractor" 7908c5d0-aa38-11ea-88d4-e341cbc94286 "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.pcap" "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286" -influx_url http://influxdb:8086 -mongo_url mongodb://mongo:27017 -s 1f487b99-8708-4d34-bcce-c0921612dcc7 -p /home/5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/profile.json
    at ChildProcess.exithandler (child_process.js:281:12)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:915:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5) 
2020-06-09T10:49:43.955Z [rest-api] info: ::ffff:127.0.0.1 - PUT /api/pcap/7908c5d0-aa38-11ea-88d4-e341cbc94286/stream/1f487b99-8708-4d34-bcce-c0921612dcc7/help HTTP/1.0 500 873 - 138.774 ms

I dug a little deeper into the system and also found the error that is thrown by `st2110_extractor`:

> [2020-06-09 10:57:22.230] [console] [error] exception: [json.exception.out_of_range.403] key 'packet_size' not found

It seems that the extractor is expecting a `packet_size` key in the posted JSON object. I checked the contents but it is not there.

Any idea what might be wrong? Is there something else I can try?

Kind regards,
Wouter

**Comment by pedro-alves-ferreira** [comment]
Hello,

Thanks for reporting this and for all the details. Can you please share the file that causes the problem with me? That would help pinpoint the issue.

Regards,

Pedro Ferreira


No dia 09/06/2020, às 13:31, woutervanrooy <notifications@github.com> escreveu:

﻿

Hello,

I am using the LIST tool for analysis of some ST2110-30 packet captures. For some of these captures, there are multiple combinations of encoding, channel count and packet time that match the packet size. LIST will then ask for additional media information so that it can complete the analysis. So far, so good.

However, whenever I try to submit one of the suggested compliant profiles the system runs into an internal server error:

2020-06-09T10:49:43.945Z [pcap-full-analysis] error: exception: Error: Command failed: "/app/bin//st2110_extractor" 7908c5d0-aa38-11ea-88d4-e341cbc94286 "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.pcap" "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286" -influx_url http://influxdb:8086 -mongo_url mongodb://mongo:27017 -s 1f487b99-8708-4d34-bcce-c0921612dcc7 -p /home/5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/profile.json
2020-06-09T10:49:43.950Z [express-middleware] error: Error: Command failed: "/app/bin//st2110_extractor" 7908c5d0-aa38-11ea-88d4-e341cbc94286 "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.pcap" "/home//5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286" -influx_url http://influxdb:8086 -mongo_url mongodb://mongo:27017 -s 1f487b99-8708-4d34-bcce-c0921612dcc7 -p /home/5edf574b9130f600390b3019/7908c5d0-aa38-11ea-88d4-e341cbc94286/profile.json
at ChildProcess.exithandler (child_process.js:281:12)
at emitTwo (events.js:126:13)
at ChildProcess.emit (events.js:214:7)
at maybeClose (internal/child_process.js:915:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
2020-06-09T10:49:43.955Z [rest-api] info: ::ffff:127.0.0.1 - PUT /api/pcap/7908c5d0-aa38-11ea-88d4-e341cbc94286/stream/1f487b99-8708-4d34-bcce-c0921612dcc7/help HTTP/1.0 500 873 - 138.774 ms

I dug a little deeper into the system and also found the error that is thrown by st2110_extractor:

[2020-06-09 10:57:22.230] [console] [error] exception: [json.exception.out_of_range.403] key 'packet_size' not found

It seems that the extractor is expecting a packet_size key in the posted JSON object. I checked the contents but it is not there.

Any idea what might be wrong? Is there something else I can try?

**Comment by woutervanrooy** [comment]
Hi Pedro,

Thanks for the quick response. The PCAP is attached:
[1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.zip](https://github.com/ebu/pi-list/files/4752448/1591697011373_7908c5d1-aa38-11ea-88d4-e341cbc94286.zip)

Kind regards,
Wouter

**Comment by pedro-alves-ferreira** [comment]
Hi Wouter,

Only now did I have time to get on this.
What is the format of the stream you sent? From the difference in RTP 
timestamps, I see that the packet time is 1ms, 48 samples per packet. 
But the RTP payload size is 240, which yields 5 bytes per sample, which 
does not make sense to me.

Thanks,

Pedro

**Comment by woutervanrooy** [comment]
Hi Pedro,

I did some digging and found out what could have been the issue. This is one of many streams captured in an automated testing environment using LIST, and apparently it is testing an unsupported AES67 encoding. The AES67 standard refers to RFC3551 clause 4.5.11 and RFC3190 clause 4 for the sample encoding. RFC3190 clause 4 does not only define L24 encoding, but also L20. This is not mentioned in the AES67 standard though.

The packet capture I sent you is an L20 / 48 kHz / 2 channels / 1 millisecond stream. I think that in the end it all comes down to a different interpretation of the standard, as I only just noticed that LIST does not have an option to select L20:
![image](https://user-images.githubusercontent.com/2813152/84386317-677dff00-abf1-11ea-8181-7851c3b7b1a5.png)

Kind regards,
Wouter

**Comment by pedro-alves-ferreira** [comment]
Ah, that explains it. Thanks for clarifying it.
Anyway, we will look into the way LIST is reporting this to make it clearer.

Regards,

Pedro


No dia 12/06/2020, às 08:38, woutervanrooy <notifications@github.com> escreveu:

﻿

Hi Pedro,

I did some digging and found out what could have been the issue. This is one of many streams captured in an automated testing environment using LIST, and apparently it is testing an unsupported AES67 encoding. The AES67 standard refers to RFC3551 clause 4.5.11 and RFC3190 clause 4 for the sample encoding. RFC3190 clause 4 does not only define L24 encoding, but also L20. This is not mentioned in the AES67 standard though.

The packet capture I sent you is an L20 / 48 kHz / 2 channels / 1 millisecond stream. I think that in the end it all comes down to a different interpretation of the standard, as I only just noticed that LIST does not have an option to select L20:
[image]<https://user-images.githubusercontent.com/2813152/84386317-677dff00-abf1-11ea-8181-7851c3b7b1a5.png>

Kind regards,
Wouter
